src/app/public/src/modules/analytics/analytics.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, NavigationEnd } from '@angular/router';
+
+import { StacheWindowRef, StacheConfigService } from '../shared';
+
+@Component({
+  selector: 'stache-analytics',
+  templateUrl: './analytics.component.html'
+})
+export class StacheAnalyticsComponent implements OnInit {
+   public constructor(
+    private windowRef: StacheWindowRef,
+    private configService: StacheConfigService,
+    private router: Router) { }
+
+  public ngOnInit(): void {
+    let gtmScript = this.windowRef.nativeWindow.document.getElementById('stache-gtm-script');
+    if (this.configService.runtime.command === 'build' && !gtmScript) {
+      this.addGoogleTagManagerScripts();
+      this.addGoogleAnalyticsMethod();
+      this.bindPageViewsToRouter();
+    }
+  }
+
+  public addGoogleTagManagerScripts(): void {
+    this.windowRef.nativeWindow.eval(`
+      (function(w,d,s,l,i){
+        w[l]=w[l]||[];w[l].push({'gtm.start':
+        new Date().getTime(),event:'gtm.js'});
+        var f=d.getElementsByTagName(s)[0],
+        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
+        j.async=true;
+        j.src='//www.googletagmanager.com/gtm.js?id='+i+dl;
+        j.id='stache-gtm-script';
+        f.parentNode.insertBefore(j,f);
+      })(window,document,'script','dataLayer','GTM-W56QP9');
+    `);
+  }
+
+  /**
+   * Because GTM adds the google analytics script to the page outside of our control,
+   * instead of loading the GA script twice, we extract the 'ga' window method out so we can
+   * subscribe to our router events and track SPA page views.
+   */
+  public addGoogleAnalyticsMethod(): void {
+    let clientId = this.configService.skyux.appSettings.stache.googleAnalyticsClientId;
+    this.windowRef.nativeWindow.eval(`
+      (function(i,r) {
+        i['GoogleAnalyticsObject']=r;
+        i[r]=i[r]||function() {
+          (i[r].q=i[r].q||[]).push(arguments)},
+          i[r].l=1*new Date();
+      })(window,'ga');
+    `);
+    this.windowRef.nativeWindow.ga('create', clientId, 'auto');
+  }
+
+  public bindPageViewsToRouter(): void {
+    this.router.events.subscribe(event => {
+      /* istanbul ignore else*/
```

Comment:
because there is no reason to test the else statement here.   We aren't writing the router event subscribe method.  or passing in a parameter for the if statement, so theres no reason to test the else/if statement here.  just that this method gets run.